providers/vm_extension_removal_policy.rb

```diff
@@ -0,0 +1,68 @@
+
+require 'json'
+
+action :set do
+  begin
+    # get the path of extension config file
+    if node['platform_family'] == 'windows'
+      extension_path = Dir["C:/Packages/Plugins/Chef.Bootstrap.WindowsAzure.ChefClient/*"].last

```

Comment:
@adamedx , the manifest file `HandlerEnvironment.json` which contains the path of the extension config is stored inside the root directory of the extension i.e. `C:/Packages/Plugins/Chef.Bootstrap.WindowsAzure.ChefClient/`. Hence, we need to know the extension's root directory for reading the manifest file. 
Using environment variable seems reasonable.